src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -262,11 +290,20 @@ public void validate() throws AppEngineException {
           "Validation Error: Java Tools jar location '"
               + JAR_LOCATIONS.get(JAVA_TOOLS_JAR) + "' is not a file.");
     }
+    try {
+      if (!isComponentInstalled("app-engine-java")) {
```

Comment:
I've just tried running `gcloud components remove app-engine-java` on my Linux machine and it did delete `~/google-cloud-sdk/platform/google_appengine/google/appengine/tools/java/lib/appengine-tools-api.jar`. When installed it back using `gcloud components install app-engine-java`, it reappeared. So, I think checking for existence of `JAVA_TOOLS_JAR` is sufficient.
@elharo, do you see a different result when you run these commands?